cmd/main.go

```diff
@@ -0,0 +1,190 @@
+	flag.StringVar(&config.TokenPath, "token-path", cfg.BuildDefaultConfigItem("TOKEN_PATH", "~/.vault-token"), "a full file path where a token will be read from/written to (env: TOKEN_PATH)")
+	flag.BoolVar(&config.K8SAuth, "k8s-auth", func() bool {
+		b, err := strconv.ParseBool(cfg.BuildDefaultConfigItem("K8S_AUTH", "false"))
+		return err == nil && b
+	}(), "select kubernetes vault auth as the vault authentication mechanism (env: K8S_AUTH)")
+	flag.BoolVar(&config.AWSAuth, "aws-auth", func() bool {
+		b, err := strconv.ParseBool(cfg.BuildDefaultConfigItem("IAM_AUTH", "false"))
+		return err == nil && b
+	}(), "select AWS IAM vault auth as the vault authentication mechanism (env: IAM_AUTH)")
+	flag.BoolVar(&config.AWSAuth, "iam-auth", func() bool {
+		b, err := strconv.ParseBool(cfg.BuildDefaultConfigItem("IAM_AUTH", "false"))
+		return err == nil && b
+	}(), "(legacy) select AWS IAM vault auth as the vault authentication mechanism (env: IAM_AUTH)")
+	flag.StringVar(&config.K8STokenPath, "k8s-token-path", cfg.BuildDefaultConfigItem("K8S_TOKEN_PATH", "/var/run/secrets/kubernetes.io/serviceaccount/token"), "kubernetes service account jtw token path (env: K8S_TOKEN_PATH)")
+	flag.StringVar(&config.VaultAuthRoleName, "vault-auth-role", cfg.BuildDefaultConfigItem("VAULT_AUTH_ROLE", ""), "the name of the role used for auth. used with either auth method (env: VAULT_AUTH_ROLE, note: will infer to k8s sa account name if left blank)")
+	flag.BoolVar(&config.GCPAuth, "gcp-auth", func() bool {
+		b, err := strconv.ParseBool(cfg.BuildDefaultConfigItem("GCP_AUTH", "false"))
+		return err == nil && b
+	}(), "select Google Cloud Platform IAM auth as the vault authentication mechanism (env: GCP_AUTH)")
+	flag.StringVar(&config.GCPServiceAccount, "gcp-svc-acct", cfg.BuildDefaultConfigItem("GCP_SVC_ACCT", ""), "the name of the service account authenticating (env: GCP_SVC_ACCT)")
+	flag.Int64Var(&config.RenewalInterval, "renewal-interval", func() int64 {
+		b, err := strconv.ParseInt(cfg.BuildDefaultConfigItem("RENEWAL_INTERVAL", "300"), 10, 64)
+		if err != nil {
+			return 900
+		}
+		return b
+	}(), "how often to check the token's ttl and potentially renew it (env: RENEWAL_INTERVAL)")
+	flag.Int64Var(&config.RenewalThreshold, "renewal-threshold", func() int64 {
+		b, err := strconv.ParseInt(cfg.BuildDefaultConfigItem("RENEWAL_THRESHOLD", "7200"), 10, 64)
+		if err != nil {
+			return 7200
+		}
+		return b
+	}(), "the threshold remaining in the vault token, in seconds, after which it should be renewed (env: RENEWAL_THRESHOLD)")
+	flag.Int64Var(&config.RenewalIncrement, "renewal-increment", func() int64 {
+		b, err := strconv.ParseInt(cfg.BuildDefaultConfigItem("RENEWAL_INCREMENT", "43200"), 10, 64)
+		if err != nil {
+			return 43200
+		}
+		return b
+	}(), "the value, in seconds, to which the token's ttl should be renewed (env: RENEWAL_INCREMENT)")
```

Comment:
Consideration: `RENEWAL_INCREMENT` isn't a value we define in the readme, and doesnt have a default. The default shouldn't be a problem since it'll leverage the TTL for  whatever the backend or role provides.